pkg/util/time.go

```diff
@@ -98,3 +98,13 @@ func NewDisableableTicker(interval time.Duration) (func(), <-chan time.Time) {
 	tick := time.NewTicker(interval)
 	return func() { tick.Stop() }, tick.C
 }
+
+type TimeRangeRequest interface {
+	GetStart() int64
+	GetEnd() int64
+}
+
+// HasTimeRange is true if the request has a start and end set.
+func HasTimeRange(req TimeRangeRequest) bool {
+	return req.GetStart() == 0 || req.GetEnd() == 0
+}
```

Comment:
I had to move this helper here, instead of adding a method directly on `typesv1.LabelNamesRequest`. If I tried to add it directly on the type, the definition got deleted every time we regenerated the protobuf code.

Reply:
Yep that make sense, may be `model` makes also sense up to you.